src/main/java/seedu/tinner/model/reminder/UniqueReminderList.java

```diff
@@ -91,19 +95,41 @@ public void add(Reminder toAdd) {
                 .observableArrayList(new TreeSet<>(dateToReminderMap.keySet())));
     }
 
+    /**
+     * Returns the list of reminders associated with a specific date.
+     *
+     * @param date The date that which is used to retrieve the list of reminders.
+     * @return the list of reminders mapped from the given date.
+     */
     public ObservableList<Reminder> getDateSpecificReminders(LocalDate date) {
         ObservableList<Reminder> reminderList = FXCollections.observableArrayList();
         ArrayList<Reminder> retrievedReminders = dateToReminderMap.get(date);
         if (retrievedReminders == null) {
             return reminderList;
         }
         Collections.sort(retrievedReminders);
-        for (Reminder reminder : retrievedReminders) {
-            reminderList.add(reminder);
-        }
+        reminderList.addAll(retrievedReminders);
         return reminderList;
     }
 
+    /**
+     * Resets the internal list of reminders using the given list of companies.
+     *
+     * @param companyList The company from which the reminders are to be produced.
+     */
+    public void setReminders(ObservableList<Company> companyList) {
+        for (Company company : companyList) {
+            ReadOnlyRoleList roleList = company.getRoleList();
+            for (Role role : roleList.getRoleList()) {
+                if (role.getReminderDate().isWithinReminderWindow()) {
+                    Reminder reminder =
+                            new Reminder(company.getName(), role.getName(), role.getStatus(), role.getReminderDate());
+                    add(reminder);
```

Comment:
Is there any way to get rid of the arrowhead here? Maybe abstracting out the if check on line 124 into another method?

Reply:
Yes that would work, I'll make the changes accordingly. Thanks!